### PR TITLE
Make router, registry, deployer use secrets

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
@pmorie can you take a look at the secrets usage to make sure I'm not doing anything insane. The interesting commit is the last one - the rest is the rebase and auth changes in other pulls I'm basing off of.

I'm decently satisfied with the registry/router/deployer setups for the moment.

The next step is to get a command that can turn a .kubeconfig or similar into a secret, for use with things like the Jenkins pod that need API credentials. The namespace defaulting would help a lot there, so our jenkins example could preset a volumemount and volume which referenced a secret named "jenkins-secret", and tell them to create such a secret in the namespace prior to creating the Jenkins pod.

**Comment by pmorie** [comment]
@liggitt started taking a look tonight on my phone, need a real screen for
a diff this robust

**Comment by liggitt** [comment]
@deads2k fyi, the last commit is necessary for registry to send identity to the API

**Comment by smarterclayton** [comment]
Jeff added just such a command upstream yesterday (as a pull).

**Comment by liggitt** [comment]
rebased

**Comment by smarterclayton** [comment]
Isn't this going to break as soon as we add service accounts?

**Comment by liggitt** [comment]
yeah, started this before we talked through service account mapping to secrets

**Comment by smarterclayton** [comment]
Up to you whether you want to land it now or hold it.  If it substantially reduces confusion I'm all for doing it now.

----- Original Message -----

File: pkg/cmd/util/clientcmd/clientcmd.go [hunk 1]
```diff
@@ -45,17 +46,64 @@ func (cfg *Config) Bind(flags *pflag.FlagSet) {
 	BindClientConfigSecurityFlags(&cfg.CommonConfig, flags)
 }
 
-func EnvVarsFromConfig(config *kclient.Config) []api.EnvVar {
-	insecure := "false"
-	if config.Insecure {
-		insecure = "true"
+func SecretVolumeFromConfig(name, namespace string, mount_path string, config *kclient.Config) (*api.Secret, *api.Volume, *api.VolumeMount, []api.EnvVar, error) {
```
Comment (pmorie) [inline pkg/cmd/util/clientcmd/clientcmd.go]: Nit: mount_path -> mountPath

**Comment by pmorie** [comment]
@smarterclayton @liggitt I don't think just service accounts would affect this - you would consume the secrets in the same way unless there was a LOAS daemon that decorated your requests to the master with auth.  Am I missing something?

**Comment by liggitt** [comment]
@pmorie we're talking about restricting a pod's access to a secret unless it is assigned a service account which also has access to the secret. double references aren't ideal, but we're having a hard time figuring out a different way to subdivide secret ACL within a namespace

File: pkg/cmd/util/clientcmd/clientcmd.go (hunk 1 above)
Comment (liggitt) [inline pkg/cmd/util/clientcmd/clientcmd.go]: Fixed

**Comment by smarterclayton** [comment]
If secret is the reference and service account is the check, then it's probably ok to put the volumes in now.  I don't think the current setup will match the experience we want for end users (expose secret when it's added to the account).

**Comment by liggitt** [comment]
closing, will rework with service accounts

**Pull request closed without merging**